pkg/apply/processor/scale.go

```diff
@@ -233,7 +234,20 @@ func (c *ScaleProcessor) MountRootfs(cluster *v2.Cluster) error {
 		return err
 	}
 
-	return fs.MountRootfs(cluster, hosts, true, false)
+	return fs.MountRootfs(cluster, hosts)
+}
+
+func (c *ScaleProcessor) Bootstrap(cluster *v2.Cluster) error {
+	logger.Info("Executing pipeline Bootstrap in CreateProcessor")
+	hosts := append(c.MastersToJoin, c.NodesToJoin...)
+	bs := bootstrap.New(cluster)
+	if err := bs.Preflight(hosts...); err != nil {
+		return err
+	}
+	if err := bs.Init(hosts...); err != nil {
+		return err
+	}
+	return bs.ApplyExternalDeps(hosts...)
```

Comment:
Is bs.ApplyExternalDeps suitable before init?

Reply:
I'm afraid not, everything needs to be initialized are in the `init.sh` process. Maybe create another hook function executed before init?

Reply:
We can extend this in the `bootstrap` module

Reply:
> We can extend this in the `bootstrap` module

Yes, all the acquired hosts should not be nodes and masters and need to be added to register, right?

Reply:
@cuisongliu I've add a `Dependency` interface, you can checkout the latest commit.